lib/ts/logger.ts

```diff
@@ -0,0 +1,49 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import debug from "debug";
+import { version } from "./version";
+
+/*
+ The debug logger below can be used to log debug messages in the following format
+    com.supertokens {t: "2022-03-18T11:15:24.608Z", message: Your message, file: "/home/supertokens-node/lib/build/supertokens.js:231:18" sdkVer: "9.2.0"} +0m
+*/
+
+function logDebugMessage(message: string) {
+    debug(`com.supertokens`)(
+        `{t: "${new Date().toISOString()}", message: \"${message}\", file: \"${getFileLocation()}\" sdkVer: "${version}"}`
+    );
+}
+
+let getFileLocation = () => {
```

Comment:
You need to make sure that this function is not called if the user has not enabled logging. Cause creating errors can be expensive.